src/main/scala/io/iohk/ethereum/ledger/LedgerUtils.scala

```diff
@@ -0,0 +1,30 @@
+package io.iohk.ethereum.ledger
+
+import scala.annotation.tailrec
+
+object LedgerUtils {
+
+  /**
+    * Function finds minimal value in some interval for which provided function do not return error
+    * If searched value is not in provided interval, function returns maximum value of searched interval
+    * @param min minimum of searched interval
+    * @param max maximum of searched interval
+    * @param f function which return error in case to little value provided
+    * @return minimal value for which provided function do not return error
+    */
+  @tailrec
+  private[ledger] def binaryChop[Error](min: BigInt, max: BigInt)(f: BigInt => Option[Error]): BigInt = {
+    assert(min <= max)
+
+    if (min == max)
+      max
+    else {
+      val mid = min + (max - min) / 2
+      val possibleError = f(mid)
+      if (possibleError.isEmpty)
```

Comment:
I think the only error we should use to continue searching is `OutOfGas` as the other ones will return the same error even if narrowing the value. I think it will converge to `max` .. wdyt?

Reply:
Actually, there are cases with other kinds of errors that will eventually converge to some minimum, good example is this one in https://github.com/input-output-hk/mantis/pull/336/files/8345e90cb532910c9ab74abd446f4fa21fa5b9f1#diff-16527700d5f0bb6dd4d24c426f5c49a3R104. If we will give it to little gas it fails with `InvalidJump` and not `OOG`
Also, other clients do not look at a type of error so I assume this is trade-off which is accepted.  (that there will be some errors that will always end with maximum gas)